File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function(tokens) {
    const stack = [];
    tokens.forEach(el => {
        if (isNaN(Number(el))) {
            const b = stack.pop();
            const a = stack.pop();
            let val;
            switch (el) {
                case "+": val = a + b; break;
                case "-": val = a - b; break;
                case "*": val = a * b; break;
                case "/": val = a / b; break;
            }
            // console.log(el);
            stack.push(Math.trunc(val));
            // console.log(stack)
        } else {
            stack.push(Number(el))
            // console.log(stack)
        }
    })
    return stack.pop();
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
import math

class Solution(object):
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        
        """
        stack = []
        operators = ["+", "-", "*", "/"]
        for item in tokens:
            if item not in operators:
                stack.append(float(item))
            else:
                y = stack.pop()
                x = stack.pop()
                val = None
                if item == '+':
                    val = x + y
                elif item == '-':
                    val = x - y
                elif item == '*':                    
                    val = x * y
                elif item == '/':
                    val = math.trunc(x / y)
                stack.append(val)
        return int(stack.pop())
        